smolyay/samples.py

```diff
@@ -0,0 +1,496 @@
+import collections.abc
+import abc
+import numpy
+
+
+class UnidimensionalPointSet(collections.abc.Sequence):
+    """Set of unidimensional points
+
+    A set of unique unidimensional points within a domain.
+
+    Parameters
+    ----------
+    domain: list
+        Domain of the sample points.
+    """
+
+    def __init__(self, domain):
+        self._points = None
+        self._domain = None
+        self._valid_cache = False
+
+        self.domain = domain
+
+    @property
+    def domain(self):
+        """numpy.ndarray: Domain of the `points`."""
+        return self._domain
+
+    @domain.setter
+    def domain(self, value):
+        domain = numpy.sort(numpy.array(value, dtype=float))
+        if domain.shape != (2,):
+            raise TypeError("Domain must be array with two variables")
+        if domain[0] == domain[1]:
```

Comment:
I think it doesn't hurt to make this `>=` if we're checking anyway:
```suggestion
        if domain[0] >= domain[1]:
```